Unable to flash M5 board -> {ModuleNotFoundError: No module named 'lark'}

### Reproduction steps

```bash
With SHA 706e9bb14f9931b70c0d4f3478a3460f56ebd099

I am unable to flash M5 board, as I am running into -> ModuleNotFoundError: No module named 'lark'

Steps to reproduce:

1. cd connectedhomeip
2. git reset --hard 706e9bb14f9931b70c0d4f3478a3460f56ebd099
3. git clean -Xdf
4. git submodule update --init
5. export PKG_CONFIG_PATH="/opt/homebrew/opt/openssl@3/lib/pkgconfig"
6. source scripts/bootstrap.sh
7. source scripts/activate.sh
8. cd ${HOME}/tools/esp-idf
9. git checkout v4.4
10. git submodule update --init --recursive
11. ./install.sh
12. . ./export.sh
13. cd ~/connectedhomeip/examples/all-clusters-app/esp32
14. idf.py build
```


### Bug prevalence

Always

### GitHub hash of the SDK that was being used

706e9bb14f9931b70c0d4f3478a3460f56ebd099

### Platform

darwin

### Platform Version(s)

_No response_

### Anything else?

Attaching the log;

[M5_error_log.txt](https://github.com/project-chip/connectedhomeip/files/9932185/M5_error_log.txt)

Note: With SHA 0c0092309188bcc5718aeb4b0f8f68eb468688ce

if I execute in the following sequence ;
1. cd connectedhomeip
2. git reset --hard 0c0092309188bcc5718aeb4b0f8f68eb468688ce
3. git clean -Xdf
4. git submodule update --init
5. export PKG_CONFIG_PATH="/opt/homebrew/opt/openssl@3/lib/pkgconfig"
6. source scripts/bootstrap.sh
7. source scripts/activate.sh
8. cd ${HOME}/tools/esp-idf
9. git checkout v4.4
10. git submodule update --init --recursive
11. ./install.sh
12. . ./export.sh
13. cd ~/connectedhomeip/examples/all-clusters-app/esp32
14. idf.py build


I get "No module named 'stringcase'" error

**complete log:**

```
[11/1258] Generating ../../gen/app-cod...p-app/app/PluginApplicationCallbacks.h
FAILED: gen/app-codegen/cpp-app/app/PluginApplicationCallbacks.h /Users/priyankahonnakasturi/connectedhomeip/examples/all-clusters-app/esp32/build/gen/app-codegen/cpp-app/app/PluginApplicationCallbacks.h 
cd /Users/priyankahonnakasturi/connectedhomeip/examples/all-clusters-app/esp32/build/esp-idf/main && /Users/priyankahonnakasturi/connectedhomeip/scripts/codegen.py --generator cpp-app --output-dir /Users/priyankahonnakasturi/connectedhomeip/examples/all-clusters-app/esp32/build/gen/app-codegen/cpp-app --expected-outputs /Users/priyankahonnakasturi/connectedhomeip/examples/all-clusters-app/esp32/build/gen/app-codegen/cpp-app/expected.outputs /Users/priyankahonnakasturi/connectedhomeip/examples/all-clusters-app/all-clusters-common/all-clusters-app.matter
Traceback (most recent call last):
  File "/Users/priyankahonnakasturi/connectedhomeip/scripts/codegen.py", line 34, in <module>
    from idl.generators import FileSystemGeneratorStorage, GeneratorStorage
  File "/Users/priyankahonnakasturi/connectedhomeip/scripts/idl/generators/__init__.py", line 23, in <module>
    from .filters import RegisterCommonFilters
  File "/Users/priyankahonnakasturi/connectedhomeip/scripts/idl/generators/filters.py", line 16, in <module>
    import stringcase
ModuleNotFoundError: No module named 'stringcase'
[22/1258] Building C object esp-idf/mb...CMakeFiles/mbedcrypto.dir/bignum.c.obj
ninja: build stopped: subcommand failed.
ninja failed with exit code 1

```

@andy31415  Following is the order of execution:
1. cd ~/tools/esp-idf
2. ./install.sh
3. . ./export.sh
4.  cd ~/connectedhomeip/examples/all-clusters-app/esp32
5. idf.py build

And this gives the modules missing error;

```
FAILED: gen/app-codegen/cpp-app/app/PluginApplicationCallbacks.h /Users/priyankahonnakasturi/connectedhomeip/examples/all-clusters-app/esp32/build/gen/app-codegen/cpp-app/app/PluginApplicationCallbacks.h 
cd /Users/priyankahonnakasturi/connectedhomeip/examples/all-clusters-app/esp32/build/esp-idf/main && /Users/priyankahonnakasturi/connectedhomeip/scripts/codegen.py --generator cpp-app --output-dir /Users/priyankahonnakasturi/connectedhomeip/examples/all-clusters-app/esp32/build/gen/app-codegen/cpp-app --expected-outputs /Users/priyankahonnakasturi/connectedhomeip/examples/all-clusters-app/esp32/build/gen/app-codegen/cpp-app/expected.outputs /Users/priyankahonnakasturi/connectedhomeip/examples/all-clusters-app/all-clusters-common/all-clusters-app.matter
Traceback (most recent call last):
  File "/Users/priyankahonnakasturi/connectedhomeip/scripts/codegen.py", line 28, in <module>
    from idl.matter_idl_parser import CreateParser
  File "/Users/priyankahonnakasturi/connectedhomeip/scripts/idl/matter_idl_parser.py", line 7, in <module>
    from lark import Lark
ModuleNotFoundError: No module named 'lark'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/priyankahonnakasturi/connectedhomeip/scripts/codegen.py", line 32, in <module>
    from idl.matter_idl_parser import CreateParser
  File "/Users/priyankahonnakasturi/connectedhomeip/scripts/idl/matter_idl_parser.py", line 7, in <module>
    from lark import Lark
ModuleNotFoundError: No module named 'lark'
[591/1243] Performing configure step for 'chip_gn'
AUTO-DETECTING input matter IDL file.
Done. Made 221 targets from 142 files in 169ms
[593/1243] Building C object esp-idf/n....dir/nghttp2/lib/nghttp2_session.c.obj
ninja: build stopped: subcommand failed.
ninja failed with exit code 1
```

> @andy31415 Following is the order of execution:
> 
> 1. cd ~/tools/esp-idf
> 2. ./install.sh
> 3. . ./export.sh
> 4. cd ~/connectedhomeip/examples/all-clusters-app/esp32
> 5. idf.py build
> 
> And this gives the modules missing error;
> 
> ```
> FAILED: gen/app-codegen/cpp-app/app/PluginApplicationCallbacks.h /Users/priyankahonnakasturi/connectedhomeip/examples/all-clusters-app/esp32/build/gen/app-codegen/cpp-app/app/PluginApplicationCallbacks.h 
> cd /Users/priyankahonnakasturi/connectedhomeip/examples/all-clusters-app/esp32/build/esp-idf/main && /Users/priyankahonnakasturi/connectedhomeip/scripts/codegen.py --generator cpp-app --output-dir /Users/priyankahonnakasturi/connectedhomeip/examples/all-clusters-app/esp32/build/gen/app-codegen/cpp-app --expected-outputs /Users/priyankahonnakasturi/connectedhomeip/examples/all-clusters-app/esp32/build/gen/app-codegen/cpp-app/expected.outputs /Users/priyankahonnakasturi/connectedhomeip/examples/all-clusters-app/all-clusters-common/all-clusters-app.matter
> Traceback (most recent call last):
>   File "/Users/priyankahonnakasturi/connectedhomeip/scripts/codegen.py", line 28, in <module>
>     from idl.matter_idl_parser import CreateParser
>   File "/Users/priyankahonnakasturi/connectedhomeip/scripts/idl/matter_idl_parser.py", line 7, in <module>
>     from lark import Lark
> ModuleNotFoundError: No module named 'lark'
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "/Users/priyankahonnakasturi/connectedhomeip/scripts/codegen.py", line 32, in <module>
>     from idl.matter_idl_parser import CreateParser
>   File "/Users/priyankahonnakasturi/connectedhomeip/scripts/idl/matter_idl_parser.py", line 7, in <module>
>     from lark import Lark
> ModuleNotFoundError: No module named 'lark'
> [591/1243] Performing configure step for 'chip_gn'
> AUTO-DETECTING input matter IDL file.
> Done. Made 221 targets from 142 files in 169ms
> [593/1243] Building C object esp-idf/n....dir/nghttp2/lib/nghttp2_session.c.obj
> ninja: build stopped: subcommand failed.
> ninja failed with exit code 1
> ```

I am saying that I specifically do NOT expect this to work: you are executing `export.sh` and that will invalidate all the python settings set up by `activate/bootstrap.sh`. That sequence of steps will not work and I don't think we can make it work in a clean matter as we have less control over what `export.sh` will set up. You found the workaround by manually installing packages, but it is a workaround.

please switch your steps by making sure export.sh is run before activate.sh and use activate.sh to setup the final build environment.

--- a/src/ble/BLEEndPoint.cpp
+++ b/src/ble/BLEEndPoint.cpp
@@ -70,17 +70,6 @@
 #define BLE_CONFIG_IMMEDIATE_ACK_WINDOW_THRESHOLD                   1
 
 /**
- * @def BLE_CONNECT_TIMEOUT_MS
- *
- * @brief
- *   This is the amount of time, in milliseconds, after a BLE end point initiates a transport protocol connection
- *   or receives the initial portion of a connect request before the end point will automatically release its BLE
- *   connection and free itself if the transport connection has not been established.
- *
- */
-#define BLE_CONNECT_TIMEOUT_MS                                15000 // 15 seconds
-
-/**
  *  @def BLE_UNSUBSCRIBE_TIMEOUT_MS
  *
  *  @brief
@@ -90,7 +79,6 @@
  */
 #define BLE_UNSUBSCRIBE_TIMEOUT_MS                            5000 // 5 seconds
 
-#define BTP_ACK_RECEIVED_TIMEOUT_MS                          15000 // 15 seconds
 #define BTP_ACK_SEND_TIMEOUT_MS                               2500 // 2.5 seconds
 
 #define BTP_WINDOW_NO_ACK_SEND_THRESHOLD                         1 // Data fragments may only be sent without piggybacked
@@ -1419,7 +1407,7 @@ bool BLEEndPoint::SendIndication(PacketBufferHandle && buf)
 CHIP_ERROR BLEEndPoint::StartConnectTimer()
 {
     const CHIP_ERROR timerErr =
-        mBle->mSystemLayer->StartTimer(System::Clock::Milliseconds32(BLE_CONNECT_TIMEOUT_MS), HandleConnectTimeout, this);
+        mBle->mSystemLayer->StartTimer(System::Clock::Milliseconds32(BTP_CONN_RSP_TIMEOUT), HandleConnectTimeout, this);
     ReturnErrorOnFailure(timerErr);
     mTimerStateFlags.Set(TimerStateFlag::kConnectTimerRunning);
 
@@ -1429,7 +1417,7 @@ CHIP_ERROR BLEEndPoint::StartConnectTimer()
 CHIP_ERROR BLEEndPoint::StartReceiveConnectionTimer()
 {
     const CHIP_ERROR timerErr =
-        mBle->mSystemLayer->StartTimer(System::Clock::Milliseconds32(BLE_CONNECT_TIMEOUT_MS), HandleReceiveConnectionTimeout, this);
+        mBle->mSystemLayer->StartTimer(System::Clock::Milliseconds32(BTP_CONN_RSP_TIMEOUT), HandleReceiveConnectionTimeout, this);
     ReturnErrorOnFailure(timerErr);
     mTimerStateFlags.Set(TimerStateFlag::kReceiveConnectionTimerRunning);
 
@@ -1440,8 +1428,8 @@ CHIP_ERROR BLEEndPoint::StartAckReceivedTimer()
 {
     if (!mTimerStateFlags.Has(TimerStateFlag::kAckReceivedTimerRunning))
     {
-        const CHIP_ERROR timerErr = mBle->mSystemLayer->StartTimer(System::Clock::Milliseconds32(BTP_ACK_RECEIVED_TIMEOUT_MS),
-                                                                   HandleAckReceivedTimeout, this);
+        const CHIP_ERROR timerErr =
+            mBle->mSystemLayer->StartTimer(System::Clock::Milliseconds32(BTP_ACK_TIMEOUT), HandleAckReceivedTimeout, this);
         ReturnErrorOnFailure(timerErr);
 
         mTimerStateFlags.Set(TimerStateFlag::kAckReceivedTimerRunning);

--- a/src/ble/BLEEndPoint.cpp
+++ b/src/ble/BLEEndPoint.cpp
@@ -1407,7 +1407,7 @@ bool BLEEndPoint::SendIndication(PacketBufferHandle && buf)
 CHIP_ERROR BLEEndPoint::StartConnectTimer()
 {
     const CHIP_ERROR timerErr =
-        mBle->mSystemLayer->StartTimer(System::Clock::Milliseconds32(BTP_CONN_RSP_TIMEOUT), HandleConnectTimeout, this);
+        mBle->mSystemLayer->StartTimer(System::Clock::Milliseconds32(BTP_CONN_RSP_TIMEOUT_MS), HandleConnectTimeout, this);
     ReturnErrorOnFailure(timerErr);
     mTimerStateFlags.Set(TimerStateFlag::kConnectTimerRunning);
 
@@ -1416,8 +1416,8 @@ CHIP_ERROR BLEEndPoint::StartConnectTimer()
 
 CHIP_ERROR BLEEndPoint::StartReceiveConnectionTimer()
 {
-    const CHIP_ERROR timerErr =
-        mBle->mSystemLayer->StartTimer(System::Clock::Milliseconds32(BTP_CONN_RSP_TIMEOUT), HandleReceiveConnectionTimeout, this);
+    const CHIP_ERROR timerErr = mBle->mSystemLayer->StartTimer(System::Clock::Milliseconds32(BTP_CONN_RSP_TIMEOUT_MS),
+                                                               HandleReceiveConnectionTimeout, this);
     ReturnErrorOnFailure(timerErr);
     mTimerStateFlags.Set(TimerStateFlag::kReceiveConnectionTimerRunning);
 
@@ -1429,7 +1429,7 @@ CHIP_ERROR BLEEndPoint::StartAckReceivedTimer()
     if (!mTimerStateFlags.Has(TimerStateFlag::kAckReceivedTimerRunning))
     {
         const CHIP_ERROR timerErr =
-            mBle->mSystemLayer->StartTimer(System::Clock::Milliseconds32(BTP_ACK_TIMEOUT), HandleAckReceivedTimeout, this);
+            mBle->mSystemLayer->StartTimer(System::Clock::Milliseconds32(BTP_ACK_TIMEOUT_MS), HandleAckReceivedTimeout, this);
         ReturnErrorOnFailure(timerErr);
 
         mTimerStateFlags.Set(TimerStateFlag::kAckReceivedTimerRunning);

--- a/src/ble/BleConfig.h
+++ b/src/ble/BleConfig.h
@@ -199,26 +199,26 @@
 
 
 /**
- * @def BTP_CONN_RSP_TIMEOUT
+ * @def BTP_CONN_RSP_TIMEOUT_MS
  *
  * @brief
  *   Maximum amount of time, in milliseconds, after sending or receiving a BTP Session Handshake
  *   request to wait for connection establishment.
  */
-#ifndef BTP_CONN_RSP_TIMEOUT
-#define BTP_CONN_RSP_TIMEOUT 15000 // 15 seconds
-#endif // BTP_CONN_RSP_TIMEOUT
+#ifndef BTP_CONN_RSP_TIMEOUT_MS
+#define BTP_CONN_RSP_TIMEOUT_MS 15000 // 15 seconds
+#endif // BTP_CONN_RSP_TIMEOUT_MS
 
 /**
- * @def BTP_ACK_TIMEOUT
+ * @def BTP_ACK_TIMEOUT_MS
  *
  * @brief
  *   Maximum amount of time, in milliseconds, after sending a BTP packet to wait for
  *   an acknowledgement. When the ack is not received within this period the BTP session is closed.
  */
-#ifndef BTP_ACK_TIMEOUT
-#define BTP_ACK_TIMEOUT 15000 // 15 seconds
-#endif // BTP_ACK_TIMEOUT
+#ifndef BTP_ACK_TIMEOUT_MS
+#define BTP_ACK_TIMEOUT_MS 15000 // 15 seconds
+#endif // BTP_ACK_TIMEOUT_MS
 
 // clang-format on
 

--- a/src/ble/BleConfig.h
+++ b/src/ble/BleConfig.h
@@ -106,16 +106,6 @@
 #endif // BLE_CONNECTION_OBJECT
 
 /**
- *  @def BLE_CONFIG_BLUEZ_MTU_FEATURE
- *
- *  @brief
- *    This define if BLUEZ MTU FEATURE is enabled or not
- */
-#ifndef BLE_CONFIG_BLUEZ_MTU_FEATURE
-#define BLE_CONFIG_BLUEZ_MTU_FEATURE 0
-#endif // BLE_CONFIG_BLUEZ_MTU_FEATURE
-
-/**
  *  @def BLE_CONNECTION_UNINITIALIZED
  *
  *  @brief
@@ -206,6 +196,29 @@
 #ifndef BLE_CONFIG_ERROR
 #define BLE_CONFIG_ERROR(e) (BLE_CONFIG_ERROR_MIN + (e))
 #endif // BLE_CONFIG_ERROR
+
+
+/**
+ * @def BTP_CONN_RSP_TIMEOUT
+ *
+ * @brief
+ *   Maximum amount of time, in milliseconds, after sending or receiving a BTP Session Handshake
+ *   request to wait for connection establishment.
+ */
+#ifndef BTP_CONN_RSP_TIMEOUT
+#define BTP_CONN_RSP_TIMEOUT 15000 // 15 seconds
+#endif // BTP_CONN_RSP_TIMEOUT
+
+/**
+ * @def BTP_ACK_TIMEOUT
+ *
+ * @brief
+ *   Maximum amount of time, in milliseconds, after sending a BTP packet to wait for
+ *   an acknowledgement. When the ack is not received within this period the BTP session is closed.
+ */
+#ifndef BTP_ACK_TIMEOUT
+#define BTP_ACK_TIMEOUT 15000 // 15 seconds
+#endif // BTP_ACK_TIMEOUT
 
 // clang-format on
 

--- a/src/transport/SecureSession.h
+++ b/src/transport/SecureSession.h
@@ -22,6 +22,7 @@
 #pragma once
 
 #include <app/util/basic-types.h>
+#include <ble/BleConfig.h>
 #include <lib/core/ReferenceCounted.h>
 #include <messaging/ReliableMessageProtocolConfig.h>
 #include <transport/CryptoContext.h>
@@ -167,9 +168,7 @@ public:
         case Transport::Type::kTcp:
             return System::Clock::Seconds16(30);
         case Transport::Type::kBle:
-            // TODO: Figure out what this should be, but zero is not the right
-            // answer.
-            return System::Clock::Seconds16(5);
+            return System::Clock::Milliseconds32(BTP_ACK_TIMEOUT);
         default:
             break;
         }

--- a/src/transport/SecureSession.h
+++ b/src/transport/SecureSession.h
@@ -168,7 +168,7 @@ public:
         case Transport::Type::kTcp:
             return System::Clock::Seconds16(30);
         case Transport::Type::kBle:
-            return System::Clock::Milliseconds32(BTP_ACK_TIMEOUT);
+            return System::Clock::Milliseconds32(BTP_ACK_TIMEOUT_MS);
         default:
             break;
         }

--- a/src/transport/UnauthenticatedSessionTable.h
+++ b/src/transport/UnauthenticatedSessionTable.h
@@ -93,7 +93,7 @@ public:
         case Transport::Type::kTcp:
             return System::Clock::Seconds16(30);
         case Transport::Type::kBle:
-            return System::Clock::Milliseconds32(BTP_ACK_TIMEOUT);
+            return System::Clock::Milliseconds32(BTP_ACK_TIMEOUT_MS);
         default:
             break;
         }

--- a/src/transport/UnauthenticatedSessionTable.h
+++ b/src/transport/UnauthenticatedSessionTable.h
@@ -16,6 +16,7 @@
  */
 #pragma once
 
+#include <ble/BleConfig.h>
 #include <lib/core/CHIPError.h>
 #include <lib/core/ReferenceCounted.h>
 #include <lib/support/CodeUtils.h>
@@ -91,6 +92,8 @@ public:
                                             GetLastPeerActivityTime(), kMinActiveTime);
         case Transport::Type::kTcp:
             return System::Clock::Seconds16(30);
+        case Transport::Type::kBle:
+            return System::Clock::Milliseconds32(BTP_ACK_TIMEOUT);
         default:
             break;
         }

